### PR TITLE
Update for Shadowlands

### PR DESCRIPTION
**Pull request by Xruptor**
I added a few missing modules and some code fixes all over the place.  Cleaned up the README and removed from leftover rFilter3 code that wasn't being used.  Added some new examples to the Shaman module of the changes.  Quite a bit of cleanup everywhere really lol.

### CODE REVIEW

**Comment by lithammer** [comment]
> Quite a bit of cleanup everywhere really lol.

No kidding 😮 It's gonna take a bit of time to review. Why is LibCustomGlow necessary? I actually have an updated version locally, just haven't bothered pushing the changes 🤷‍♂️

File: libs/LibStub/tests/test4.lua [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+-- Again with a huge number
+LibStub.minor = LibStub.minor + 1234567890
+
+loadfile("../LibStub.lua")()
+
+assert(LibStub.IterateLibraries == 123)
+
+
+print("OK")
```
Comment (lithammer) [inline libs/LibStub/tests/test4.lua]: Remove the `libs/LibStub/tests` directory.

File: tempenchants.lua [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	end
+	
+	if self.hideOutOfCombat and not InCombatLockdown() then
+		alpha = 0
+	end
+	
+	return alpha or 0
+end
+
+local function SetGlow(self, alpha)
```
Comment (lithammer) [inline tempenchants.lua]: This function is defined four times.

File: cooldowns.lua [hunk 3]
```diff
@@ -1,87 +1,145 @@
 local _, addon = ...
+local cfg = addon.cfg
 
-local globalCooldown = 1.5
+local LibGlow = LibStub("LibCustomGlow-1.0")
+local decimalThreshold = 3
 
 local function GetAlpha(self, duration)
```
Comment (lithammer) [inline cooldowns.lua:4]: This defined four times.

**Comment by Xruptor** [comment]
> > Quite a bit of cleanup everywhere really lol.
> 
> No kidding 😮 It's gonna take a bit of time to review. Why is LibCustomGlow necessary? I actually have an updated version locally, just haven't bothered pushing the changes 🤷‍♂️

LibGlow adds the ability to have additional custom action glow animations.  It's also used by WeakAuras.  It's a neat little library.

**Comment by Xruptor** [comment]
> > Quite a bit of cleanup everywhere really lol.
> 
> No kidding 😮 It's gonna take a bit of time to review. Why is LibCustomGlow necessary? I actually have an updated version locally, just haven't bothered pushing the changes 🤷‍♂️

Oh in that case, if you would like to use your updated version, I'll retract this pull request before making anymore changes. 😄

**Comment by lithammer** [comment]
> > > Quite a bit of cleanup everywhere really lol.
> > 
> > 
> > No kidding 😮 It's gonna take a bit of time to review. Why is LibCustomGlow necessary? I actually have an updated version locally, just haven't bothered pushing the changes 🤷‍♂️
> 
> Oh in that case, if you would like to use your updated version, I'll retract this pull request before making anymore changes. 😄

My changes are only bare minimum (fixing errors) and updated auras for some classes. So I'll definitely take this for a spin when I have some time, since it's seems like you've made some nice improvements 🙂 I'm actually surprised anyone was using this (except me of course).

**Comment by Xruptor** [comment]
I searched the internet for an addon that was a bare minimum of WeakAuras.  I find the code and options of WeakAuras to be extremely bloated and really only wanted something really simple to track stuff.  I was surprised to find that there are virtually almost no addons out there that do this except yours.  Which I came across.  So I decided to update the addon and provide the missing functionality that I needed.  It worked out great. 😄

File: libs/LibStub/tests/test4.lua (hunk 1 above)
Comment (Xruptor) [reply libs/LibStub/tests/test4.lua]: I'll remove them.

File: cooldowns.lua (hunk 3 above)
Comment (Xruptor) [reply cooldowns.lua:4]: I see that's my bad.  The old copy and paste syndrome got me.

Comment (Xruptor) [reply cooldowns.lua:4]: Consolidated the code.

File: tempenchants.lua (hunk 2 above)
Comment (Xruptor) [reply tempenchants.lua]: Consolidated the code.

**Comment by Xruptor** [comment]
There seems to be a slight bug with the Spells module while tracking if is Usable and on Cooldown.  I believe the global cooldown is causing issues.  I will test it a bit more tomorrow.  I think a combination of IsUsableSpell and GetSpellCooldown is required to circumvent this.  I'll do some trial and error testing later lol.

**Comment by Xruptor** [comment]
Yep it's what I thought.  It was a pesky global cooldown issue.

**Comment by lithammer** [comment]
Sorry for being slow. Noticed my subscription had ended (haven't logged in for many weeks). Gonna get that sorted.

**Pull request closed without merging**